test/spec/utils_spec.js

```diff
@@ -922,4 +922,34 @@ describe('Utils', function () {
       });
     });
   });
+
+  describe('mergeDeep', function() {
+    it('properly merge objects that share same property names', function() {
+      const object1 = {
+        propA: {
+          subPropA: 'abc'
+        }
+      };
+      const object2 = {
+        propA: {
+          subPropB: 'def'
+        }
+      };
+      
+      const resultWithoutMergeDeep = Object.assign({}, object1, object2);
+      expect(resultWithoutMergeDeep).to.deep.equal({
+        propA: {
+          subPropB: 'def'
+        }
+      });
+
+      const resultWithMergeDeep = utils.mergeDeep({}, object1, object2);
+      expect(resultWithMergeDeep).to.deep.equal({
+        propA: {
+          subPropA: 'abc',
+          subPropB: 'def'
+        }
+      });
+    });
+  });
```

Comment:
I think you need some more tests dealing with other property types besides plain objects, also merging objects of different depths.

Reply:
@leonardlabat can you please add more test cases as suggested by @snapwich ?

Reply:
Just added 2 test cases (and support for merging arrays)
Tell me if you think it still need more tests